modules/openapi-generator/src/main/resources/kotlin-client/data_class.mustache

```diff
@@ -74,7 +74,7 @@ import {{packageName}}.infrastructure.ITransformForStorage
 @Deprecated(message = "This schema is deprecated.")
 {{/isDeprecated}}
 {{>additionalModelTypeAnnotations}}
-{{#nonPublicApi}}internal {{/nonPublicApi}}{{#discriminator}}interface{{/discriminator}}{{^discriminator}}data class{{/discriminator}} {{classname}}{{^discriminator}} (
+{{#nonPublicApi}}internal {{/nonPublicApi}}{{#discriminator}}interface{{/discriminator}}{{^discriminator}}{{#hasVars}}data {{/hasVars}}class{{/discriminator}} {{classname}}{{^discriminator}} (
```

Comment:
@jaqxues the `hasVars` is true when the model has at least one property.

what I don't understand is there's no change to the samples (e.g. kotlin petstore client) given that the test petstore pet has at least one model with properties.

Reply:
This fixes an issue with models without properties, so the opposite case. A `data class` in kotlin is not allowed to have no properties

Reply:
Thanks for the clarification 👍